airsonic-main/src/test/java/org/airsonic/player/dao/AlbumDaoTest.java

```diff
@@ -0,0 +1,155 @@
+import org.airsonic.player.domain.MusicFolder;
+import org.airsonic.player.service.search.AbstractAirsonicHomeTest;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.util.ObjectUtils;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class AlbumDaoTest extends AbstractAirsonicHomeTest {
+
+    private List<MusicFolder> musicFolders;
+
+    @Autowired
+    private MediaFileDao mediaFileDao;
+
+    @Autowired
+    private AlbumDao albumDao;
+
+    private List<MusicFolder> sameName;
+    private List<MusicFolder> noTagFirst;
+    private List<MusicFolder> nameInReverse;
+
+    private final int offset = 0;
+    private final int count = Integer.MAX_VALUE;
+
+    @Before
+    public void setup() {
+        if (ObjectUtils.isEmpty(musicFolders)) {
+            sameName = Arrays.asList(new MusicFolder(1, new File(resolveBaseMediaPath.apply("Scan/ID3Album/sameName")), "1", true, new Date()));
+            noTagFirst = Arrays.asList(new MusicFolder(2, new File(resolveBaseMediaPath.apply("Scan/ID3Album/noTagFirst")), "2", true, new Date()));
+            nameInReverse = Arrays.asList(new MusicFolder(3, new File(resolveBaseMediaPath.apply("Scan/ID3Album/nameInReverse")), "3", true, new Date()));
+            musicFolders = new ArrayList<>();
+            musicFolders.addAll(sameName);
+            musicFolders.addAll(noTagFirst);
+            musicFolders.addAll(nameInReverse);
+        }
+        populateDatabaseOnlyOnce();
+    }
+
+    @Override
+    public List<MusicFolder> getMusicFolders() {
+        return musicFolders;
+    }
+
+    @Test
+    public void testUpdateAlbum() {
+
+        // File structure
+        List<MediaFile> albums = mediaFileDao.getAlphabeticalAlbums(offset, count, false, sameName);
+        assertEquals(1, albums.size());
+        MediaFile album = albums.get(0);
+        assertEquals("ALBUM1", album.getName());
+        assertEquals("albumArtistA", album.getArtist());
+        assertNull(album.getAlbumArtist());
+        assertEquals("genreA", album.getGenre());
+        assertEquals(Integer.valueOf(2001), album.getYear());
+
+        // ID3 tags
+        List<Album> albumId3s = albumDao.getAlphabeticalAlbums(offset, count, false, false, sameName);
+        assertEquals(1, albumId3s.size());
+        Album albumId3 = albumId3s.get(0);
+
+        // [albumA albumArtistA genreB 2002]
+        // This combination is "a combination that should not exist.
+        assertEquals("albumA", albumId3.getName());
+        assertEquals("albumArtistA", albumId3.getArtist());
+        assertEquals("genreB", albumId3.getGenre()); // XXX Unintentional overwrite
+        assertEquals(Integer.valueOf(2002), albumId3.getYear()); // XXX Unintentional overwrite
```

Comment:
The following records are generated in the album table.

``albumA/albumArtistA/genreB/2002``

Real files do not have this combination.

Reply:
Does this really happen with the current scanner? That's bad...

Reply:
The fix method that has relatively little effect is as follows :

The following processing ....

https://github.com/airsonic/airsonic/blob/f930a8e06bd5d87f66f5713620f347576499588a/airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java#L292-L300

Put in this block.

https://github.com/airsonic/airsonic/blob/f930a8e06bd5d87f66f5713620f347576499588a/airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java#L306-L313

 - However, including other properties, you need to think carefully and add tests.
 - It's hard to tell if this is a simple bug fix. Of course, data inconsistencies can simply be eliminated. However, there are doubts as to whether this is the expected data specification, and honestly it is a blank page.

Currently, the number of processes that depend on ID3 is extremely small, but it is not without it.

https://github.com/airsonic/airsonic/blob/f930a8e06bd5d87f66f5713620f347576499588a/airsonic-main/src/main/java/org/airsonic/player/service/upnp/AlbumUpnpProcessor.java#L81-L94

``// TODO: correct artist?``

A little unclear.
Note that you should be aware of these effects if you change the Album data specification rather than eliminating inconsistencies.
Sonos doesn't take a good look.